src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionQRInfoViewController.swift

```diff
@@ -71,8 +68,30 @@ class ExposureSubmissionQRInfoViewController: DynamicTableViewController, ENANav
 		view.backgroundColor = .enaColor(for: .background)
 		cellBackgroundColor = .clear
 
+		tableView.register(
+			UINib(nibName: String(describing: DynamicAcknowledgementCell.self), bundle: nil),
+			forCellReuseIdentifier: ReuseIdentifiers.acknowledgement.rawValue
+		)
+
+		tableView.register(
+			UINib(nibName: String(describing: LabeledCountriesCell.self), bundle: nil),
+			forCellReuseIdentifier: ReuseIdentifiers.countries.rawValue
+		)
+
 		dynamicTableViewModel = viewModel.dynamicTableViewModel
 		tableView.separatorStyle = .none
 	}
+}
+
+// MARK: - Protocol ENANavigationControllerWithFooterChild
+extension ExposureSubmissionQRInfoViewController: ENANavigationControllerWithFooterChild {
 
+	func navigationController(_ navigationController: ENANavigationControllerWithFooter, didTapPrimaryButton button: UIButton) {
+		onPrimaryButtonTap { [weak self] isLoading in
+			DispatchQueue.main.async {
+				self?.navigationFooterItem?.isPrimaryButtonLoading = isLoading
+				self?.navigationFooterItem?.isPrimaryButtonEnabled = !isLoading
+			}
+		}
+	}
```

Comment:
why put this in an extension? It's basically a protocol implementation. Shouldn't we move this up in the class implementation?

Reply:
Style. I've seen and adopted this pattern over time to put protocol implementations into extensions and keep them together. Better than MARKs and cluttered functions & properties.